### PR TITLE
Failure recovery.

### PR DESCRIPTION
**Pull request by DingtongHan**
Failure detection and recovery, see more details in docs/Monitor_proposal.md

### CODE REVIEW

**Comment by lgarithm** [comment]
you can run `go fmt ./...` to format all the go code. (don't change go.mod)

File: srcs/go/utils/iostream/iostream.go [hunk 1]
```diff
@@ -19,6 +20,10 @@ func Tee(r io.Reader, ws ...io.Writer) error {
 		}
 		for _, w := range ws {
 			fmt.Fprintln(w, string(line))
+			datas := strings.Split(string(line), ":")
```
Comment (lgarithm) [inline srcs/go/utils/iostream/iostream.go:23]: This packages is for generic utils functions, they shouldn't contain application logics.

File: srcs/go/libkungfu-comm/cmds.go [hunk 2]
```diff
@@ -2,14 +2,93 @@ package main
...
+        },
+    }
 //export GoKungfuRunMain
 func GoKungfuRunMain() {
 	args := os.Args[1:] // remove wrapper program name (`which python`)
 	app.Main(args)
 }
+
+//export GoKungfuRunSendBegin
+func GoKungfuRunSendBegin() {
```
Comment (lgarithm) [inline srcs/go/libkungfu-comm/cmds.go:32]: you can abstract one API for the monitoring signal, and use different data to represent different signal types.

Comment (DingtongHan) [reply srcs/go/libkungfu-comm/cmds.go:35]: Have changed

**Comment by DingtongHan** [comment]
> you can run `go fmt ./...` to format all the go code. (don't change go.mod)

File: srcs/go/libkungfu-comm/cmds.go [hunk 3]
```diff
@@ -1,15 +1,65 @@
...
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
 )
 
 import "C"
 
+var serverip string
```
Comment (lgarithm) [inline srcs/go/libkungfu-comm/cmds.go:18]: could you move the changes to a new file and leave `cmds.go` unchanged.

Comment (DingtongHan) [reply srcs/go/libkungfu-comm/cmds.go]: Done

**Review by lgarithm — APPROVED** [review]

**Pull request merged**